src/output/render/permissions.rs

```diff
@@ -76,6 +88,33 @@ impl f::Permissions {
     }
 }
 
+impl f::Attributes {
+    pub fn render<C: Colours+FiletypeColours>(&self, colours: &C) -> Vec<ANSIString<'static>> {
+        let bit = |bit, chr: &'static str, style: Style| {
+            if bit { style.paint(chr) }
+              else { colours.dash().paint("-") }
+        };
+
+        vec![
+            bit(self.archive,   "a", colours.normal()),
+            bit(self.readonly,  "r", colours.user_read()),
+            bit(self.hidden,    "h", colours.special_user_file()),
+            bit(self.system,    "s", colours.special_other()),
+        ]
+    }
+
+    pub fn render_type<C: Colours+FiletypeColours>(&self, colours: &C) -> ANSIString<'static> {
```

Comment:
This is to mimic the behaviour of PowerShell.

Reply:
Not sure what you mean by that? Not sure what this code does, but it’s worthy of a comment on GitHub, maybe it’s worthy to explain the thing quickly in a comment (at least it’s not obvious what’s going on for a non-Powershell user, I’d say).

Reply:
Oh i mean this is to mimic the behaviour of `gci` command in PowerShell, which is the builtin `ls` alternative in PowerShell.